metagraph/plugins/pandas/types.py

```diff
@@ -223,16 +251,43 @@ def assert_equal(
                 g1 = obj1.value
                 g2 = obj2.value
                 assert len(g1) == len(g2), f"{len(g1)} != {len(g2)}"
-                assert len(obj1.index & obj2.index) == len(
-                    obj1.index
-                ), f"{len(obj1.index & obj2.index)} != {len(obj1.index)}"
+
+                if aprops1["is_directed"]:
+                    assert len(obj1.index & obj2.index) == len(
+                        obj1.index
+                    ), f"edge mismatch {obj1.index ^ obj2.index}"
+                else:  # undirected
```

Comment:
The suggestion from https://github.com/metagraph-dev/metagraph/pull/139/files#r520790460 would also apply here.